lib/network/modules/components/nodes/util/NodeBase.js

```diff
@@ -89,6 +90,23 @@ class NodeBase {
       }
     }
   }
+
+
+  /**
+   * Determine if the shape of a node needs to be recalculated.
+   *
+   * @protected
+   */
+  needRefresh(selected, hover) {
```

Comment:
There are no guidelines. I don't know the code good enough to know if there is a schema for this. Have a look at the surrounding code - you decide.